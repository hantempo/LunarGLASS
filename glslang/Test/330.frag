#version 330 compatibility

in vec4 inVar;
out vec4 outVar;

varying vec4 varyingVar;

void main()
{
    gl_FragColor = varyingVar;  // link ERROR: user output was used
    gl_FragData[1] = inVar;     // link ERROR: user output was used
    int buffer = 4;
}

#extension GL_ARB_separate_shader_objects : enable

in gl_PerFragment {
    vec4 gl_Color;
};

void foo()
{
    vec4 c = gl_Color;
    outVar = inVar;
}
