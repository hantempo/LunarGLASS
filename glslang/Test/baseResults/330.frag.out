330.frag
Warning, version 330 is not yet complete; some version-specific features are present, but many are missing.

0:? Sequence
0:8  Function Definition: main( (void)
0:8    Function Parameters: 
0:10    Sequence
0:10      move second child to first child (4-component vector of float)
0:10        'gl_FragColor' (fragColor 4-component vector of float)
0:10        'varyingVar' (smooth in 4-component vector of float)
0:11      move second child to first child (4-component vector of float)
0:11        direct index (4-component vector of float)
0:11          'gl_FragData' (fragColor 32-element array of 4-component vector of float)
0:11          Constant:
0:11            1 (const int)
0:11        'inVar' (smooth in 4-component vector of float)
0:12      Sequence
0:12        move second child to first child (int)
0:12          'buffer' (int)
0:12          Constant:
0:12            4 (const int)
0:21  Function Definition: foo( (void)
0:21    Function Parameters: 
0:23    Sequence
0:23      Sequence
0:23        move second child to first child (4-component vector of float)
0:23          'c' (4-component vector of float)
0:23          gl_Color: direct index for structure (in 4-component vector of float)
0:23            '__anon__0' (in block{gl_Color})
0:23            Constant:
0:23              0 (const uint)
0:24      move second child to first child (4-component vector of float)
0:24        'outVar' (out 4-component vector of float)
0:24        'inVar' (smooth in 4-component vector of float)
0:?   Linker Objects
0:?     'inVar' (smooth in 4-component vector of float)
0:?     'outVar' (out 4-component vector of float)
0:?     'varyingVar' (smooth in 4-component vector of float)
0:?     '__anon__0' (in block{gl_Color})


Linked fragment stage:

ERROR: Linking fragment stage: Cannot use gl_FragColor or gl_FragData when using user-defined outputs
ERROR: Linking fragment stage: Cannot use both gl_FragColor and gl_FragData


